index.d.ts

```diff
@@ -0,0 +1,9 @@
+type ClassValue = string | number | ClassDictionary | ClassArray;
+
+interface ClassDictionary {
+	[id: string]: boolean;
+}
+
+interface ClassArray extends Array<ClassValue> { }
+
+export default function(...classes: ClassValue[]): string;
```

Comment:
both `import classNames from 'classNames';` and `import * as classNames from 'classNames';` are generally OK. The last one that uses `require()` is discouraged because it is old CommonJS-style syntax.

but, like I said, the DefinitelyTyped README stipulates that you should not declare a default export unless `require("classnames").default` actually returns the module.